File: src/main/java/me/kosik/interwalled/ailist/AIListConfig.java
package me.kosik.interwalled.ailist;


public class AIListConfig {
    final int maximumComponentsCount;
    final int intervalsCountToCheckLookahead;
    final int intervalsCountToTriggerExtraction;
    final int minimumComponentSize;

    public AIListConfig(
            final int maximumComponentsCount,
            final int intervalsCountToCheckLookahead,
            final int intervalsCountToTriggerExtraction,
            final int minimumComponentSize
    ) {
        this.maximumComponentsCount = maximumComponentsCount;
        this.intervalsCountToCheckLookahead = intervalsCountToCheckLookahead;
        this.intervalsCountToTriggerExtraction = intervalsCountToTriggerExtraction;
        this.minimumComponentSize = minimumComponentSize;
    }

    static public AIListConfig DEFAULT =
        new AIListConfig(10, 20, 10, 64);
}

File: ailist/src/main/scala/me/kosik/interwalled/ailist/AIListConfig.scala
package me.kosik.interwalled.ailist

case class AIListConfig(
  maximumComponentsCount: Int,
  intervalsCountToCheckLookahead: Int,
  intervalsCountToTriggerExtraction: Int,
  minimumComponentSize: Int,
  isInputDataSorted: Boolean
)

object AIListConfig {
  val DEFAULT = AIListConfig(10, 20, 10, 64, isInputDataSorted = false)
}